### PR TITLE
fix reflection deprecated php 8.0 methods

### PR DESCRIPTION
**Pull request by lampatrampa**
Fix deprecated reflection from issue #53

### CODE REVIEW

**Comment by wolfy-j** [comment]
@SerafimArts sounds like Travis broke PHP version again and it fails to pull the coverage. Can you take a look?

**Comment by SerafimArts** [comment]
@wolfy-j yep. What should I do?

**Comment by wolfy-j** [comment]
Maybe it makes sense to migrate to GA or update Travis to use an embedded PHP engine for coverage.

File: src/Method.php [hunk 1]
```diff
@@ -64,9 +64,12 @@ public static function match(\ReflectionMethod $method): bool
         if ($method->getNumberOfParameters() != 2) {
             return false;
         }
+        
+        $ctxType = $method->getParameters()[0]->getType();
+        $inType = $method->getParameters()[1]->getType();
 
-        $ctx = $method->getParameters()[0]->getClass();
-        $in = $method->getParameters()[1]->getClass();
+        $ctx = $ctxType && ! $ctxType->isBuiltin() ? new \ReflectionClass($ctxType->getName()) : null;
```
Comment (DanRaduDragomir) [inline src/Method.php:69]: isBuiltIn() has been removed in PHP 8

**Comment by wolfy-j** [comment]
@SerafimArts

**Comment by DanRaduDragomir** [comment]
While at it, maybe adding either a private function or a general wrapper lib dealing with this ain't a bad idea
something along the lines of:

```php
class classReflector
{
    /**
     * @param ReflectionMethod $method
     * @return array
     * @throws ReflectionException
     */
    public static function getMethodPropertyTypes(ReflectionMethod $method): array
    {
        $types = [];
        foreach ($method->getParameters() as $key => $parameter) {
            $types[$key] = self::getMethodPropertyType($parameter);
        }
        return $types;
    }

    /**
     * @param ReflectionParameter $parameter
     * @return ReflectionClass|null
     * @throws ReflectionException
     */
    public static function getMethodPropertyType(ReflectionParameter $parameter): ReflectionClass|null
    {
        $type =  $parameter->getType();
        return $type && $type instanceof \ReflectionNamedType && !$type->isBuiltin() ? new \ReflectionClass($type->getName()) : null;
    }
}
```
just my 2c since I know there was the same issue in the framework and I think it's probably best to add this as a lib or something to be used in both places. 
Also can be easily deprecated/dropped when the time comes.

File: src/Method.php [hunk 2]
```diff
@@ -110,7 +113,7 @@ public static function parse(\ReflectionMethod $method): Method
 
         $m = new self();
         $m->name = $method->getName();
-        $m->input = $method->getParameters()[1]->getClass()->getName();
+        $m->input = $method->getParameters()[1]->getType()->getName();
```
Comment (DanRaduDragomir) [inline src/Method.php:114]: This might be caught as a polymorphic call by code analyzers. Maybe use the same syntax as the one proposed above?

**Pull request merged**